src/ResultSet.php

```diff
@@ -7,9 +7,10 @@
 use Amp\Iterator;
 use Amp\Producer;
 use Amp\Promise;
+use Amp\Sql\Operation;
 use Amp\Success;
 
-final class ResultSet implements Iterator, Operation {
+final class ResultSet implements Operation, \Amp\Sql\ResultSet {
```

Comment:
\Amp\Sql\ResultSet should directly extend Operation instead of having to implement both interfaces here